### PR TITLE
Upgrade to Scala.js 1.10.1.

### PR DESCRIPTION
**Pull request by sjrd**
It is fine to do this in a patch version, because

* The IR version was not changed in 1.10.x.
* There was no change in the scalajs-library API.

---

I'm not sure who's available for reviews, these days. Anyone up for it?

### CODE REVIEW

**Comment by sjrd** [comment]
@mbovel Would you be available to review this PR, perhaps?

File: compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala [hunk 1]
```diff
@@ -1829,7 +1829,15 @@ class JSCodeGen()(using genCtx: Context) {
         }
 
       case Block(stats, expr) =>
-        js.Block(stats.map(genStat) :+ genStatOrExpr(expr, isStat))
+        // #15419 Collapse { <undefined-param>; BoxedUnit } to <undefined-param>
+        val genStatsAndExpr0 = stats.map(genStat(_)) :+ genStatOrExpr(expr, isStat)
```
Comment (mbovel) [inline compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala:1833]: Minor, style: did you add the underscore on purpose?

```suggestion
        val genStatsAndExpr0 = stats.map(genStat) :+ genStatOrExpr(expr, isStat)
```

**Review by mbovel — APPROVED** [review]
Looks great to me! 😄

File: compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala (hunk 1 above)
Comment (sjrd) [reply compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala:1833]: Yes, I always add the underscore on purpose. For some reason I don't trust eta expansion :sweat_smile:

**Comment by sjrd** [comment]
> Would that make sense to add a test for this?

We get the test for free in the upgrade to Scala.js 1.10.1. ;) Since it's part of the test suite of Scala.js 1.10.1.

**Comment by mbovel** [comment]
> > Would that make sense to add a test for this?
> 
> We get the test for free in the upgrade to Scala.js 1.10.1. ;) Since it's part of the test suite of Scala.js 1.10.1.

Oh, I see.

**Pull request merged**